ibm_db_django/pybase.py

```diff
@@ -321,14 +321,18 @@ def execute( self, operation, parameters = () ):
                 if operation.count( "%s" ) > 0:
                     operation = operation.replace("%s", "?")
 
-            # Ensure SELECT statements have WITH NC to read locked records.
+            # Ensure SELECT statements have WITH NC and USE CURRENTLY COMMITTED to read locked records.
+            # - https://www.ibm.com/docs/en/i/7.4?topic=statement-isolation-clause
+            # - https://www.ibm.com/docs/en/i/7.4?topic=statement-concurrent-access-resolution-clause
             # update final sql before execute so we can read locked records from iseries db
             # sql is original generated from as_sql in SQL compilier
             # but the generated sql may not be final, looks like it still get processed in several places
             is_select_statement = operation.startswith('SELECT') or operation.startswith('WITH')
             # `WITH` at start also indicates it's a SELECT statement, since the WITH is the start of a CTE (Common Table Expressions), and CTEs can only be used in SELECT statements
             if is_select_statement and ' WITH NC' not in operation:
                 operation = operation + ' WITH NC'
+            if is_select_statement and ' WITH NC' not in operation:
```

Comment:
isnt this code unreachable? 
line 333 we update `operation` to include ` WITH NC `, then now line 334 we run the same check again and `'WITH NC' not in operation` check will be always false

Reply:
@duongpc Yes it is. Sorry I corrected it in my local working copy that I was testing with in mtdirect2-prototype (I mounted the file in that project), but forgot to correct it here. This is the version I was testing with:
![image](https://github.com/user-attachments/assets/8d996aec-8d35-4c8c-92d6-bf30a55307af)
Fixed in 33aac3db751be3dfaa4a05629f78dde2d88b8d65